src/cpp/redis.cpp

```diff
@@ -49,50 +49,29 @@ Redis::~Redis()
         delete this->_redis;
 }
 
-CommandReply Redis::run(Command& cmd)
-{
-    Command::iterator cmd_fields_start = cmd.begin();
-    Command::iterator cmd_fields_end = cmd.end();
-    CommandReply reply;
-
-    int n_trials = 100;
-
-    while (n_trials > 0) {
+CommandReply Redis::run(SingleKeyCommand& cmd){
+    Command* bcmd = &cmd;
+    return _run(*bcmd);
```

Comment:
``_run`` should be ``this->_run``.  Also, do we need to create the pointer and then dereference the pointer?  Is there an easier to read and elegant way to call a common ``_run`` function for all of these ``run`` fuctions?